command/agent/consul/client.go

```diff
@@ -566,6 +566,10 @@ func (c *ServiceClient) RegisterAgent(role string, services []*structs.Service)
 			Tags:    service.Tags,
 			Address: host,
 			Port:    port,
+			// This enables the consul UI to show that Nomad registered this service
+			Meta: map[string]string{
+				"external-source": "nomad",
```

Comment:
@preetapan would it be possible to have the nomad alloc ID in this meta key set as well? would be some of my work 1000x easier at work when trying to map nomad service{} to consul service id outside of nomad